src/zenml/integrations/tensorflow/visualizers/tensorboard_visualizer.py

```diff
@@ -0,0 +1,216 @@
+            "...to visualize the Tensorboard logs for your trained model."
+        )
+
+    def stop(
+        self,
+        object: StepView,
+    ) -> None:
+        """Stop the Tensorboard server previously started for a pipeline step.
+
+        Args:
+            object: StepView fetched from run.get_step().
+        """
+        for _, artifact_view in object.outputs.items():
+            # filter out anything but model artifacts
+            if artifact_view.type == ModelArtifact.TYPE_NAME:
+                logdir = os.path.dirname(artifact_view.uri)
+
+                # first check if a Tensorboard server is already running for
+                # the same logdir location and use that one
+                running_server = self.find_running_tensorboard_server(logdir)
+                if not running_server:
+                    return
+
+                logger.debug(
+                    "Stopping tensorboard server with PID '%d' ...",
+                    running_server.pid,
+                )
+                try:
+                    p = psutil.Process(running_server.pid)
+                except psutil.Error:
+                    logger.error(
+                        "Could not find process for PID '%d' ...",
+                        running_server.pid,
+                    )
+                    continue
+                p.kill()
+                return
+
+
+def visualize_tensorboard(pipeline_name: str, step_name: str) -> None:
+    """Start a Tensorboard server to visualize all models logged as output by
+    the named pipeline step. The server will monitor and display all the models
+    logged by past and future step runs.
+
+    Args:
+        pipeline_name: the name of the pipeline
+        step_name: pipeline step name
+    """
+    repo = Repository()
+    pipeline = repo.get_pipeline(pipeline_name)
+    if pipeline is None:
+        raise RuntimeError(f"No pipeline with name `{pipeline_name}` was found")
+
+    last_run = pipeline.runs[-1]
+    step = last_run.get_step(name=step_name)
+    if step is None:
+        raise RuntimeError(
+            f"No pipeline step with name `{step_name}` was found in "
+            f"pipeline `{pipeline_name}`"
+        )
+    TensorboardVisualizer().visualize(step)
+
+
+def stop_tensorboard_server(pipeline_name: str, step_name: str) -> None:
+    """Stop the Tensorboard server previously started for a pipeline step.
+
+    Args:
+        pipeline_name: the name of the pipeline
+        step_name: pipeline step name
+    """
+    repo = Repository()
+    pipeline = repo.get_pipeline(pipeline_name)
+    if pipeline is None:
+        raise RuntimeError(f"No pipeline with name `{pipeline_name}` was found")
+
+    last_run = pipeline.runs[-1]
+    step = last_run.get_step(name=step_name)
+    if step is None:
+        raise RuntimeError(
+            f"No pipeline step with name `{step_name}` was found in "
+            f"pipeline `{pipeline_name}`"
+        )
```

Comment:
This can probably be extracted into a separate function as it's identical with the code above right?